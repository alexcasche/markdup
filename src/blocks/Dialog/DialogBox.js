import styled from 'styled-components'
import { tint } from 'polished'

const DialogBox = styled.div`
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  margin-top: 60px;
  padding: 30px;
  width: calc(100% - 40px);
  max-width: 600px;
  text-align: right;
  background-color: ${props => tint(0.85, props.theme.colors.black)};
  box-shadow: 0 2px 10px 0 ${props => props.theme.colors.black};
  border-radius: 5px;
`

export default DialogBox
