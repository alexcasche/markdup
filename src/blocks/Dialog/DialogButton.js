import { Button } from 'elements'
import { tint } from 'polished'

const DialogButton = Button.extend`
  display: inline-block;
  padding: 10px 18px;
  background-color: ${props => tint(0.78, props.theme.colors.black)};
  color: ${props => tint(0.65, props.theme.colors.grey)};
  font-size: 14px;
  border-radius: 2px;
  margin-left: 15px;
  &:hover {
    background-color: ${props => tint(0.74, props.theme.colors.black)};
    color: ${props => tint(0.25, props.theme.colors.grey)};
  }
`

export default DialogButton
