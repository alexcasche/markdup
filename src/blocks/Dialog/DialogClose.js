import { Button } from 'elements'
import { tint } from 'polished'

const DialogClose = Button.extend`
  background-color: ${props => tint(0.92, props.theme.colors.black)};
  position: absolute;
  top: 0;
  right: 0;
  padding: 8px 11px 8px 12px;
  border-radius: 0 5px 0 5px;
  font-size: 16px;
  svg {
    color: ${props => tint(0.65, props.theme.colors.grey)};
  }
  &:hover {
    background-color: ${props => tint(0.94, props.theme.colors.black)};
    svg {
      color: ${props => tint(0.6, props.theme.colors.grey)};
    }
  }
`

export default DialogClose
