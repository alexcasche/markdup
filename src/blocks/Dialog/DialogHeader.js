import styled from 'styled-components'
import { tint } from 'polished'

const DialogHeader = styled.h1`
  margin: 0;
  text-align: left;
  color: ${props => tint(0.25, props.theme.colors.grey)};
  font-weight: bold;
  font-size: 18px;
  letter-spacing: 1px;
  line-height: 28px;
`

export default DialogHeader
