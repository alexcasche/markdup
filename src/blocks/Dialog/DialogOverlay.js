import styled from 'styled-components'

const DialogOverlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: ${props => props.theme.colors.black};
  opacity: 0.5;
  z-index: -1;
`

export default DialogOverlay
