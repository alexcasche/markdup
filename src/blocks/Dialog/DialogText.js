import styled from 'styled-components'
import { tint } from 'polished'

const DialogText = styled.p`
  margin: 0;
  padding: 20px 20px 30px;
  text-align: left;
  color: ${props => tint(0.85, props.theme.colors.grey)};
  font-size: 17px;
  font-weight: 500;
  letter-spacing: 1px;
  line-height: 1.75;
  svg {
    color: ${props => props.theme.colors.green};
    margin-right: 10px;
  }
`

export default DialogText
