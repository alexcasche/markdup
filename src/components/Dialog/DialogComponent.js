import React from 'react'
import { createPortal } from 'react-dom'
import { Dialog } from 'blocks'
import { default as DialogInner } from './dialogFactory'
import { PropTypes as t } from 'prop-types'

Dialog.Inner = DialogInner

const DialogComponent = props => {
  if (props.showDialog) {
    return createPortal(
      <Dialog className="Dialog">
        <Dialog.Overlay />
        <Dialog.Inner {...props} />
      </Dialog>,
      document.getElementById('modal-root')
    )
  } else return null
}

DialogComponent.propTypes = {
  currentWorkspace: t.shape({
    html: t.string.isRequired,
    markdown: t.string.isRequired,
    title: t.string.isRequired,
  }).isRequired,
  numWorkspaces: t.number.isRequired,
  showDialog: t.oneOf(['deleteWorkspace', 'resetSession', 'saveSession', false])
    .isRequired,
  showSave: t.bool.isRequired,
  actions: t.shape({
    deleteWorkspace: t.func.isRequired,
    resetSession: t.func.isRequired,
    saveSession: t.func.isRequired,
    updateLayout: t.func.isRequired,
  }).isRequired,
}

export default DialogComponent
