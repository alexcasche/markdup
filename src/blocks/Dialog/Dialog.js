import styled from 'styled-components'

import DialogBox from './DialogBox'
import DialogButton from './DialogButton'
import DialogClose from './DialogClose'
import DialogHeader from './DialogHeader'
import DialogOverlay from './DialogOverlay'
import DialogText from './DialogText'

const Dialog = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

Dialog.Box = DialogBox
Dialog.Button = DialogButton
Dialog.Close = DialogClose
Dialog.Header = DialogHeader
Dialog.Overlay = DialogOverlay
Dialog.Text = DialogText

export default Dialog
