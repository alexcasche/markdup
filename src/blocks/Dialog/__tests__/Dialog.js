import Dialog from '../Dialog'

describe('Dialog Block', () => {
  it('renders correctly', () => {
    const tree = render(<Dialog />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
