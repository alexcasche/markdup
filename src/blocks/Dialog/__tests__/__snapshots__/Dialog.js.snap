// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Block renders correctly 1`] = `
.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

<div
  className="c0"
/>
`;
